src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/model/OffenderEmployment.java

```diff
@@ -0,0 +1,158 @@
+import javax.persistence.Column;
+import javax.persistence.Convert;
+import javax.persistence.Embeddable;
+import javax.persistence.EmbeddedId;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@Entity
+@NoArgsConstructor
+@AllArgsConstructor
+@Table(name = "OFFENDER_EMPLOYMENTS")
+public class OffenderEmployment {
+
+    @EmbeddedId
+    private PK id;
+
+    @Column(name = "EMPLOYMENT_DATE")
+    private LocalDate startDate;
+
+    @Column(name = "TERMINATION_DATE")
+    private LocalDate endDate;
+
+    @Column(name = "EMPLOYMENT_POST_CODE")
+    @Enumerated(EnumType.STRING)
+    private EmploymentPostType postType;
+
+    @Column(name = "EMPLOYER_NAME")
+    private String employerName;
+
+    @Column(name = "SUPERVISOR_NAME")
+    private String supervisorName;
+
+    @Column(name = "POSITION")
+    private String position;
+
+    @Column(name = "TERMINATION_REASON_TEXT")
+    private String terminationReason;
+
+    @Column(name = "WAGE")
+    private Double wage;
+
+    @Column(name = "WAGE_PERIOD_CODE")
+    private PayPeriodType wagePeriod;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumnsOrFormulas(value = {
+        @JoinColumnOrFormula(formula = @JoinFormula(value = "'" + Occupation.DOMAIN + "'", referencedColumnName = "domain")),
+        @JoinColumnOrFormula(column = @JoinColumn(name = "OCCUPATIONS_CODE", referencedColumnName = "code"))
+    })
+    private Occupation occupation;
+
+    @Column(name = "COMMENT_TEXT")
+    private String comment;
+
+    @Column(name = "SCHEDULE_TYPE")
+    private ScheduleType scheduleType;
+
+    @Column(name = "HOURS_WEEK")
+    private Integer hoursWeek;
+
+    @Column(name = "EMPLOYER_AWARE_FLAG")
+    @Convert(converter = YesNoToBooleanConverter.class)
+    private Boolean isEmployerAware;
+
+    @Column(name = "CONTACT_EMPLOYER_FLAG")
+    @Convert(converter = YesNoToBooleanConverter.class)
+    private Boolean isEmployerContactable;
+
+
+    @Where(clause = "OWNER_CLASS = '" + OffenderEmploymentAddress.ADDR_TYPE + "'")
+    @OneToMany(mappedBy = "employment", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private List<OffenderEmploymentAddress> addresses = new ArrayList<>();
+
+    @Data
+    @Embeddable
+    @EqualsAndHashCode
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class PK implements Serializable {
+        @Column(name = "OFFENDER_BOOK_ID")
+        private Long bookingId;
+
+        @Column(name = "EMPLOY_SEQ")
+        private Long employSeq;
+    }
+
+    @Getter
+    @RequiredArgsConstructor
+    public enum EmploymentPostType {
+        CARE("Carer"),
```

Comment:
Do we do this else where? Isn't this data, reference data that can possibly change?

Reply:
yes, should really be reference data I think